plugins/inmemoryscanner/test/Scanner_unittest.cpp

```diff
@@ -0,0 +1,262 @@
+#include "../src/Scanner.h"
+#include "FakeYara.h"
+#include "mock_Config.h"
+#include "mock_Dumping.h"
+#include "mock_PluginInterface.h"
+#include "mock_Yara.h"
+#include <gmock/gmock-matchers.h>
+#include <gmock/gmock.h>
+#include <gtest/gtest.h>
+
+using testing::_;
+using testing::An;
+using testing::ByMove;
+using testing::ContainsRegex;
+using testing::NiceMock;
+using testing::Return;
+using testing::Unused;
+
+class ScannerTestBaseFixture : public testing::Test
+{
+  protected:
+    size_t maxScanSize = 0x3200000;
+    pid_t testPid = 4;
+
+    std::unique_ptr<MockPluginInterface> pluginInterface = std::make_unique<MockPluginInterface>();
+    std::shared_ptr<MockConfig> configuration = std::make_shared<MockConfig>();
+    std::optional<Scanner> scanner;
+
+    std::string uidRegEx = "[0-9]+";
+    std::string vmiResultsOutputDir = "vmiResultsOutput";
+    std::string inMemOutputDir = "inMemDumps";
+    std::string dumpedRegionsDir = "dumpedRegions";
+
+    std::string protectionFlagsString = "RWX";
+    ProtectionValues protectionFlags = ProtectionValues::PAGE_GUARD_PAGE_EXECUTE_READWRITE;
+
+    std::filesystem::path inMemoryDumpsPath = std::filesystem::path(vmiResultsOutputDir) / inMemOutputDir;
+    std::filesystem::path dumpedRegionsPath = inMemoryDumpsPath / dumpedRegionsDir;
+    virtual_address_t startAdress = 0x1234000;
+    size_t size = 0x666;
+
+    void SetUp() override
+    {
+        ON_CALL(*pluginInterface, getResultsDir()).WillByDefault([]() { return std::make_unique<std::string>(); });
+        ON_CALL(*configuration, getMaximumScanSize()).WillByDefault(Return(maxScanSize));
+        // make sure that we return a non-empty memory region or else we might skip important parts
+        ON_CALL(*pluginInterface, readProcessMemoryRegion(_, _, _))
+            .WillByDefault([]() { return std::make_unique<std::vector<uint8_t>>(6, 9); });
+
+        ON_CALL(*pluginInterface, getResultsDir())
+            .WillByDefault([vmiResultsOutputDir = vmiResultsOutputDir]()
+                           { return std::make_unique<std::string>(vmiResultsOutputDir); });
+        ON_CALL(*configuration, getOutputPath())
+            .WillByDefault([inMemOutputDir = inMemOutputDir]() { return std::string(inMemOutputDir + "/"); });
```

Comment:
should we disable this codacy-check, rename the variables or is there an even better way?

Reply:
afaik this is a false positive. IIRC capturing like this was recommended in _Effective Modern C++_ for some reason, but I don't have the book here anymore.